utils/network.go

```diff
@@ -225,22 +223,24 @@ func setupRoutes(hostVeth netlink.Link, result *current.Result) error {
 				// exactly what we are intending to program.
 				// If the route we want is already there then most likely it's programmed by Felix, so we ignore it,
 				// and we return an error if none of the routes match the route we're trying to program.
+				log.WithFields(log.Fields{"route": route, "scope": route.Scope}).Debug("Constructed route")
 				for _, r := range routes {
-					if reflect.DeepEqual(r, route) {
+					log.WithFields(log.Fields{"interface": hostVeth.Attrs().Name, "route": r, "scope": r.Scope}).Debug("Routes for the interface")
+					if r.LinkIndex == route.LinkIndex && r.Dst.IP.Equal(route.Dst.IP) && r.Scope == route.Scope {
 						// Route was already present on the host.
-						log.Infof("CNI skipping add route. Route already exists for %s\n", hostVeth.Attrs().Name)
+						log.WithFields(log.Fields{"interface": hostVeth.Attrs().Name}).Infof("CNI skipping add route. Route already exists")
 						return nil
 					}
 				}
-				return fmt.Errorf("route (Dst: %s, Scope: %v) already exists for an interface other than '%s'",
-					route.Dst.String(), route.Scope, hostVeth.Attrs().Name)
+				return fmt.Errorf("route (Ifindex: %d, Dst: %s, Scope: %s) already exists for an interface other than '%s'",
```

Comment:
Why the change from `%v` to `%s` here and below; I think scope isn't a string so `%s` will fail.  `%v` always works so it's a sensible default unless you know you want something different.

Reply:
good catch, fixed it for both PRs